src/olympia/abuse/models.py

```diff
@@ -840,6 +841,14 @@ class CantBeAppealed(Exception):
     pass
 
 
+class CinderPolicyQuerySet(models.QuerySet):
+    def without_parents_if_their_children_are_present(self):
+        parents_ids = set(
+            self.filter(parent__isnull=False).values_list('parent', flat=True)
+        )
```

Comment:
I've got a concern that by evaluating `self.filter` at this point it will mean that `without_parents_if_their_children_are_present` _has_ to be last in a chain of `exclude`|`filter`or `parents_ids` will be wrong.  Is there a way to write this without the it? 😬